auxiliary/scripts/pull-descriptions.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env python3
+
+#------------------------------------------------------------------------
+# Pull Pattern subclass descriptions from their pycode formatting
+#------------------------------------------------------------------------
+
+import re
+import os
+import glob
+import argparse
+
+def main(args):
+    class_data = parse_class_data(args)
+    if args.generate_markdown:
+        generate_index(class_data)
+        generate_class_pages(class_data)
+    else:
+        for file_dict in class_data:
+            print(file_dict["name"])
+            for class_dict in file_dict["classes"]:
+                print(class_dict)
+            print()
+
+
+def generate_index(class_data):
+    """
+    Iterates over each of the source files indexed in `class_data`, and outputs
+    a single page of Markdown which is the documentation index.
+    """
+    contents = ""
+    for file_dict in class_data:
+        contents += "## %s\n" % file_dict["name"].title()
+        contents += "View source: [%s.py](https://github.com/ideoforms/isobar/tree/master/isobar/pattern/%s.py)\n\n" % (
+            file_dict["name"], file_dict["name"])
+        contents += "| Class | Function |\n"
+        contents += "|-|-|\n"
+        for class_dict in file_dict["classes"]:
+            contents += "| [%s](%s/%s.md) | %s |\n" % (class_dict["classname"],
+                                                       file_dict["name"],
+                                                       class_dict["classname"].lower(),
+                                                       class_dict["short_description"])
+        contents += "\n\n"
+    # Output to library.md
+    f = open("docs/patterns/library.md", "w")
+    f.write(contents)
+    f.close()
+
+def generate_class_pages(class_data):
+    for file_dict in class_data:
+        # Check if folder exists
+        if not os.path.exists("docs/patterns/%s" % file_dict['name'].lower()):
```

Comment:
Better in general to use `os.path.join` to combine path elements - create a `root_dir` variable containing `docs/patterns`, and then use `os.path.join` to create pathnames under this.

Reply:
Did for this and two other instances where we were combining `docs/patterns` with some variable. There are other paths still lingering (ie. `with open("docs/patterns/library.md", "w") as f:`) but I imagine there isn't much benefit to swapping that out with `os.path.join`, right?